### DISCUSSION TITLE
How to create a custom EBSD data file and import that data in MTEX

### DISCUSSION THREAD

**Discussion by vivek9626** [mtex-toolbox/mtex#1358] [Ask Anything] [answered]
Hello all, 

I am new to MTEX and I am trying to create a custom EBSD dataset that I want MTEXX to import but I am not sure in what format should I create the custom data. For now, I have information about the pixel Euler angles, pixel coordinates, pixel phases (Just a single-phase SrTiO3), pixel feature ids (These are information that I extracted from a 2D slice of a 3D HEDM data (.dream3D file)) but I don't know in what format should I save these data as in should I save these in a text file in a specific order and save it in a specific directory for the import wizard to work? 

I would appreciate it if anyone can provide me with some guidance or a resource that provides me information on how to make custom ebsd data and import it into mtex. I looked through the MTEX website and found something in the link below
https://mtex-toolbox.github.io/EBSDImport.html

In this link, it states something about you can find the example of writing your own interface in the directory MTEX/qta/interfaces. However, I am not able to find the directory MTEX/qta/interfaces in my windows PC so I am not sure what to do.

Please let me know information on how to go about creating the custom ebsd file which I can import in MTEX.

Thank you for your help.

Regards,
Vivek

**Comment by nyyssont** [comment, marked as answer]
I think you could make a csv file and save it as a text file, then use loadEBSD_generic. See here:

https://mtex-toolbox.github.io/loadEBSD_generic.html

Do you think you could work with this?

**Reply by vivek9626** [reply]
I will give it a try and let you know how it works

**Reply by vivek9626** [reply]
Works fine, thANKS

**Reply by Sarah-ol** [reply]
Hello, 

I tried to do that as well but can't make it work... Could you help me out please?
I'm working on olivine grains, and here's the code I'm using:

fname = fullfile(mtexDataPath,'EBSD','test2.csv');
CS = crystalSymmetry('mmm', [4.756 10.207 5.98], 'mineral', 'Forsterite');
SS = specimenSymmetry('orthorhombic');
ebsd = loadEBSD_csv(fname,'CS',CS,'SS',SS, 'ColumnNames',{'x','y','Euler1','Euler2','Euler3','phase'}, 'Bunge');

My "test2" csv file has the following columns: 'x', 'y', 'Euler1', 'Euler2', 'Euler3', 'phase'.

Thank you!
Sarah
